src/machine/machine_rp2040_pwm.go

```diff
@@ -256,21 +253,34 @@ func (pwm *pwmGroup) setDivMode(mode uint32) {
 // setPeriod sets the pwm peripheral period (frequency). Calculates DIV_INT and sets it from following equation:
 //  cycles = (TOP+1) * (CSRPHCorrect + 1) * (DIV_INT + DIV_FRAC/16)
 // where cycles is amount of clock cycles per PWM period.
-func (pwm *pwmGroup) setPeriod(period uint64) {
-	targetPeriod := uint32(period)
-	periodPerCycle := getPeriod()
+func (pwm *pwmGroup) setPeriod(period uint32) {
+	periodPerCycle := cpuPeriod()
 	top := pwm.getWrap()
 	phc := pwm.getPhaseCorrect()
-	_, frac := pwm.getClockDiv()
+	// We first modify TOP if desired period is outside divider threshold bounds.
```

Comment:
I did it the other way around for other implementations. I intentionally do not change the divider but instead only change the TOP value, because that's usually far more precise.